tests/appsec/test_asm_standalone.py

```diff
@@ -81,9 +75,10 @@ def test_no_appsec_upstream__no_attack__is_kept_with_priority_1__from_minus_1(se
         spans_checked = 0
         tested_meta = {"_dd.p.appsec": None, "_dd.p.other": "1"}
         tested_metrics = {"_sampling_priority_v1": lambda x: x < 2}
+        rid = get_rid_from_request(self.r)
 
         for data, trace, span in interfaces.library.get_spans(request=self.r):
-            if not REQUESTDOWNSTREAM_RESOURCE_PATTERN.search(span["resource"]):
+            if not get_rid_from_span(span) == rid:
```

Comment:
I'm quite surprised, because `get_spans(request=self.r)` returns only spans where `get_rid_from_span(span) == rid`. So the next `if` is redundant.

Reply:
Then I am not sure that I understand the purpose of `REQUESTDOWNSTREAM_RESOURCE_PATTERN.search(span["resource"])`, as (except for Ruby) the spans that it returns also contains rid. But it will not work with Ruby as the resource name will not look like "GET /requestdownstream". Removing it does not break any test in any library

Reply:
cc @gnufede or @iunanua ?

Reply:
As long as we're checking the spans for the downstream request, it is fine by me

Reply:
How can we be sure we're checking the request to `GET /requestdownstream` and not some other span?

Reply:
> How can we be sure we're checking the request to `GET /requestdownstream` and not some other span?

In Ruby, we don't respect the 'GET /requestdownstream' spec for now (it looks like SystemTests#RequestDownstream) so we don't want to check against 'GET /requestdownstream'.
But from my understanding, 'GET /requestdownstream' spans for every libraries (and SystemTests#RequestDownstream for Ruby) are the spans that contains the rid in their user-agent tag.

Reply:
> I'm quite surprised, because `get_spans(request=self.r)` returns only spans where `get_rid_from_span(span) == rid`. So the next `if` is redundant.

Fixed in eca34bdb0c2fd984cbff9f225656d8aa296adaeb